### PR TITLE
GDAL 3.5.1-2 & MapServer 8.0.0 on EL9

### PR DESCRIPTION
**Pull request by jbronn**
* Integrate OSGeo/gdal#6105 into a GDAL 3.5.1-2 release.
* Upgrade MapServer to 8.0.0-0 ([beta2](https://github.com/MapServer/MapServer/releases/tag/rel-8-0-0-beta2)), and support prereleases like we've done for GEOS & PostGIS.

### CODE REVIEW

**Review by BuechnerGIS — APPROVED** [review]
Good stuff! :rocket:

**Comment by pedros007** [comment]
Thanks!  I'll use MapServer 8, but not quite yet.  MapServer-8 has some minor breaking changes I need to experiment with.  We'll probably do it once it's out of beta.

File: docs/provenance.el9.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ and are [MIT licensed](./licenses/Fedora-LICENSE):
 
 The [mapserver](https://github.com/MapServer/MapServer) source archives are obtained directly from its [GitHub releases](https://github.com/MapServer/MapServer/releases).
 
-The [`mapserver.spec`](../SPECS/el9/mapserver.spec) and [`mapserver_php8.patch`](../SOURCES/el9/mapserver_php8.patch) files originate from [Fedora's `mapserver` RPM](https://src.fedoraproject.org/rpms/mapserver) and are released under the [Fedora license](./licenses/Fedora-LICENSE).
+The [`mapserver.spec`](../SPECS/el9/mapserver.spec) file originate from [Fedora's `mapserver` RPM](https://src.fedoraproject.org/rpms/mapserver) and are released under the [Fedora license](./licenses/Fedora-LICENSE).
```
Comment (hummeltech) [inline docs/provenance.el9.md]: Minor typo.
```suggestion
The [`mapserver.spec`](../SPECS/el9/mapserver.spec) file originates from [Fedora's `mapserver` RPM](https://src.fedoraproject.org/rpms/mapserver) and is released under the [Fedora license](./licenses/Fedora-LICENSE).
```

**Review by hummeltech — APPROVED** [review]
Awesome

**Comment by hummeltech** [comment]
Awesome, [MapServer 8.0.0-rc1](https://github.com/MapServer/MapServer/releases/tag/rel-8-0-0-rc1) has since been released.

**Comment by jbronn** [comment]
@hummeltech updated to use RC1 in [c57acd9](https://github.com/radiant-maxar/geoint-deps/pull/64/commits/c57acd9ac774d228b6e49ed6f5926356a9fc4624).  GDAL 3.5.1-2 and MapServer 8.0.0-0 are signed & published.

**Pull request merged**